baremetal/metal3machine_manager_test.go

```diff
@@ -563,6 +594,96 @@ var _ = Describe("Metal3Machine manager", func() {
 		)
 	})
 
+	type testCaseSetPauseAnnotation struct {
+		BMMachine     *capm3.Metal3Machine
+		Host          *bmh.BareMetalHost
+		ExpectPresent bool
+		ExpectError   bool
+	}
+
+	DescribeTable("Test Set BMH Pause Annotation",
+		func(tc testCaseSetPauseAnnotation) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), tc.Host, tc.BMMachine)
+			machineMgr, err := NewMachineManager(c, nil, nil, nil, tc.BMMachine, klogr.New())
+			Expect(err).NotTo(HaveOccurred())
+
+			err = machineMgr.SetPauseAnnotation(context.TODO())
+			if tc.ExpectError {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			savedHost := bmh.BareMetalHost{}
+			err = c.Get(context.TODO(),
+				client.ObjectKey{
+					Name:      tc.Host.Name,
+					Namespace: tc.Host.Namespace,
+				},
+				&savedHost,
+			)
+			Expect(err).NotTo(HaveOccurred())
+			if tc.ExpectPresent {
+				Expect(savedHost.Annotations[bmh.PausedAnnotation]).NotTo(BeNil())
+			} else {
+				Expect(savedHost.Annotations[bmh.PausedAnnotation]).To(BeNil())
+				return
```

Comment:
this return is unneeded (it's the end of the function anyways)

Reply:
Done